### PR TITLE
Adds latest still supported ubuntu lts templates

### PR DESCRIPTION
**Pull request by madalinignisca**
I wanted to add a few more ubuntu lts templates that I use often on local work I do and thought it would be useful for others to have them available.

### CODE REVIEW

File: examples/ubuntu-22.04.yaml
Comment (AkihiroSuda) [inline examples/ubuntu-22.04.yaml:1]: Should we now `mv ubuntu.yaml ubuntu-24.04.yaml; ln -s ubuntu-24.04.yaml ubuntu.yaml` for consistency?

File: examples/ubuntu-20.04.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# This template requires Lima v0.7.0 or later.
+images:
+# Try to use release-yyyyMMdd image if available. Note that release-yyyyMMdd will be removed after several months.
+- location: "https://cloud-images.ubuntu.com/releases/20.04/release-20240612/ubuntu-20.04-server-cloudimg-amd64.img"
+  arch: "x86_64"
+  digest: "sha256:7fce49f6c9afd2cdd9c26618b056f8806e500dd4d6eff516c9c6b5fab12e7f79"
+- location: "https://cloud-images.ubuntu.com/releases/20.04/release-20240612/ubuntu-20.04-server-cloudimg-arm64.img"
```
Comment (AkihiroSuda) [inline examples/ubuntu-20.04.yaml:7]: It is hard to keep every image up-to-date, so we may need to have an automation script before adding new templates
- https://github.com/lima-vm/lima/issues/1347

File: examples/ubuntu-22.04.yaml
Comment (madalinignisca) [reply examples/ubuntu-22.04.yaml:1]: Would be ok

**Comment by afbjorklund** [comment]
Ubuntu 16.04 and 18.04 are probably old enough to hide them from the general public (even if supported)

Having 20.04 and 22.04 should be old enough I think? And then we don't have to re-live all old bugs, like:

* #2387

Should probably delete centos-7.yaml (after tomorrow) as well, even if it is from 2014 rather than 2018...

**Comment by madalinignisca** [comment]
Sorry, but my intention with this PR is only to introduce common ubuntu templates, not change anything not having a relation with them.

I agree automation to ensure latest releases would be great, but even in my case, an update after deployment is not such a pain, as usually even in public cloud environments, full upgrade is part of my cloud-init setup (even public cloud ain't having latest and greatest releases).

For next step, I'm willing when I get some time off for some automation for some of the templates that concern me, being Debian and Ubuntu, as I don't use any other distributions for server work.

**Comment by AkihiroSuda** [comment]
We now have an updater for Ubuntu (https://github.com/lima-vm/lima/blob/master/hack/update-template-ubuntu.sh), so this PR can be revisited again after rebasing.

File: examples/deprecated/ubuntu-18.04.yaml
Comment (AkihiroSuda) [inline examples/deprecated/ubuntu-18.04.yaml:1]: This should be probably removed due to EOL

**Comment by AkihiroSuda** [comment]
Opened an alternative PR:
- https://github.com/lima-vm/lima/pull/2795

**Pull request closed without merging**